js/checkout.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+class Blockonomics {
+
+    constructor({
+        checkout_id = 'blockonomics_checkout'
+    }={}) {
+
+        // User Params
+        this.checkout_id = checkout_id
+
+        this.data = {}
+
+        // Computed Properties
+        this.progress = {}
+    }
+
+    init() {
+        this.container = document.getElementById(this.checkout_id);
+        if (!this.container) {
+            throw Error(`Blockonomics Initialisation Error: Container #${this.checkout_id} was not found!`)
+        }
+
+        if (!window.blockonomics_data) {
```

Comment:
This check should not be required since we also check for a valid JSON object later: https://github.com/blockonomics/woocommerce-plugin/pull/276/files#diff-12c8849e0a66d1a1666bb20d82b2a73b44af2208c029260cfe9db50b35515666R75-R79
The valid JSON check should rather be moved into the init function as it makes more sense to define `this.data = JSON.parse(blockonomics_data)` in init() rather than create_bindings().

Reply:
Updated based on this.